### PR TITLE
Move QBO from LedgerSync

### PR DESCRIPTION
**Pull request by jozefvaclavik**
Imports codebase from `ledger_sync` and changes namespace from `LedgerSync::Ledgers::QuickBooksOnline` to `LedgerSync::QuickBooksOnline`

### CODE REVIEW

**Comment by jozefvaclavik** [comment]
```Finished in 13.71 seconds (files took 2.8 seconds to load)
183 examples, 0 failures```

**Comment by jozefvaclavik** [comment]
Got QA working.
```
Finished in 1 minute 19.4 seconds (files took 2.76 seconds to load)
21 examples, 0 failures
```

**Comment by jozefvaclavik** [comment]
Both spec and QA are green with 1.7.0.

```
Finished in 13.32 seconds (files took 2.97 seconds to load)
204 examples, 0 failures, 21 pending

Finished in 1 minute 30.25 seconds (files took 2.8 seconds to load)
21 examples, 0 failures
```

**Comment by jozefvaclavik** [comment]
Rubocop is green as well
```Inspecting 270 files
..............................................................................................................................................................................................................................................................................

270 files inspected, no offenses detected```

**Review by amouxkin — APPROVED** [review]

File: lib/ledger_sync/quickbooks_online/deserializer.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        merge_resources_for_full_update(
+          hash: hash,
+          ledger_resource: deserialized_resource,
+          resource: resource
+        )
+      end
+
+      def merge_resources_for_full_update(args = {})
+        hash            = args.fetch(:hash)
+        ledger_resource = args.fetch(:ledger_resource)
```
Comment (amouxkin) [inline lib/ledger_sync/quickbooks_online/deserializer.rb:36]: `ledger_resource` is redeclared in the next few lines, so this isn't used.

**Review by amouxkin — APPROVED** [review]

File: lib/ledger_sync/quickbooks_online/deserializer.rb (hunk 1 above)
Comment (jozefvaclavik) [reply lib/ledger_sync/quickbooks_online/deserializer.rb:36]: Nice catch!

Comment (ryanwjackson) [reply lib/ledger_sync/quickbooks_online/deserializer.rb:36]: @jozefvaclavik can you resolve this and then let's get this merged?

Comment (jozefvaclavik) [reply lib/ledger_sync/quickbooks_online/deserializer.rb]: @amouxkin @ryanwjackson Done.

**Pull request merged**